package.json

```diff
@@ -7,7 +7,8 @@
     "fast-base64-decode": "^1.0.0"
   },
   "peerDependencies": {
-    "react-native": ">=0.56"
+    "react-native": "^0.61.5",
+    "react-native-windows": "^0.61.15"
```

Comment:
Hmmm, do we really need to change the base react-native version here?

Also, now npm/yarn will complain about missing peer dependencies for people not using react-native-windows? Is it best practice to list it as a peer-dependency?

Reply:
The base version shouldn't be changed just for windows.  Yes when using react-native-windows the required version will be >=0.61, but we shouldn't push that on all platforms.  Otherwise to use android/ios you'd suddenly need 0.61 which isn't true.  We should just document the requirement in the README.md or something.

The peer version for react-native-windows should be >=0.61.5, not ^ -- otherwise it won't work for 0.62 etc.

Unfortunately there isn't a npm concept of optionalPeerDependency or something.  The build of the windows module will require using react-native-windows, so it is a dependency when using the windows platform.

Reply:
Really, you don't even need to make react-native-windows a peerDependency if you don't want. At the end of the day, it's an app's dependency on rnw that will pull the package down.

Reply:
That's not quite always right though. If a peer dependency isn't declared, the package manager won't neccesarily put the 'react-native-windows' package in a place that's reachable by this package. Some package managers like PNPM explicitly create a special node_modules structure so that anything not declared as a dependency cannot be found.

So, if you were missing a peer dep, then were using PNPM, logic in the vcxproj to find react-native-windows wouldn't be able to find anything.

Similarly, you can have a dependency tree with two different versions of RNW, and if you don't have a peer dependency, a version of react-native-windows might not be made visible to this module, even if you're using yarn/npm.

Basically, it works 90% of the time, but will fail in certain cases if you don't include it.